File: docs/snippets/all/archetypes/video_simple.py
"""Log a video file."""

import sys

import rerun as rr

if len(sys.argv) < 2:
    print(f"Usage: {sys.argv[0]} <path_to_video.[mp4]>")
    sys.exit(1)

rr.init("rerun_example_asset_video", spawn=True)

rr.log(f"world/video", rr.AssetVideo(path=sys.argv[1]))

File: docs/snippets/all/archetypes/video_simple.rs
//! Log a video file.

use rerun::external::anyhow;

fn main() -> anyhow::Result<()> {
    let args = std::env::args().collect::<Vec<_>>();
    let Some(path) = args.get(1) else {
        anyhow::bail!("Usage: {} <path_to_video.[mp4]>", args[0]);
    };

    let rec = rerun::RecordingStreamBuilder::new("rerun_example_asset_video").spawn()?;

    rec.log("world/video", &rerun::AssetVideo::from_file_path(path)?)?;

    Ok(())
}
